querydsl-core/src/main/java/com/querydsl/core/DefaultQueryMetadata.java

```diff
@@ -306,6 +301,9 @@ public void setLimit(Long limit) {
 
     @Override
     public void setModifiers(@Nullable QueryModifiers restriction) {
+        if (restriction == null) {
+            throw new NullPointerException();
```

Comment:
Why is it marked as `@Nullable`, but must be nonnull? I think that it makes more sense to use `QueryModifiers.EMPTY` if null is specified, than to throw a `NullPointerException`.

Also, if this annotation must carry meaning, consider moving this to the interface as well.